src/results.cpp

```diff
@@ -518,6 +518,65 @@ TableView Results::get_tableview()
     REALM_UNREACHABLE();
 }
 
+static std::vector<size_t> parse_keypath(StringData keypath, Schema const& schema, const ObjectSchema *object_schema)
+{
+    auto check = [&](bool condition, const char* fmt, auto... args) {
+        if (!condition) {
+            throw std::invalid_argument(util::format("Cannot sort on key path '%1': %2.",
+                                                     keypath, util::format(fmt, args...)));
+        }
+    };
+    auto sortable_type = [](PropertyType type) {
+        return !is_array(type) && type != PropertyType::LinkingObjects && type != PropertyType::Data;
+    };
+
+    const char* begin = keypath.data();
+    const char* end = keypath.data() + keypath.size();
+    check(begin != end, "missing property name");
+    check(std::find(begin, end, '@') == end, "sorting on collection operators is not implemented");
```

Comment:
I suppose there's no benefit to doing the check here rather than in the obj-c code calling it since it's just a contains check and not something that requires looking at the parsed form.